os/arch/arm/src/armv7-a/arm_cpupause.c

```diff
@@ -417,4 +417,53 @@ int up_cpu_resume(int cpu)
 	return OK;
 }
 
+/****************************************************************************
+ * Name: up_cpu_pause_all
+ *
+ * Description:
+ *   pause all the CPUs other than the current cpu. Internally, it calls 
+ * 	 up_cpu_pause() api to stop all the CPU's.
+ *
+ * Returned Value:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int up_cpu_pause_all()
+{
+	int me = sched_getcpu();
+	for (int cpu = 0; cpu < CONFIG_SMP_NCPUS; cpu++) {
+		if (cpu != me) {
```

Comment:
We can add the condition in for statement:
`for (int cpu = 0; cpu != me && cpu < CONFIG_SMP_NCPUS; cpu++) {`

Reply:
I think, it will get fail when me = CPU0, it will exit from the loop and won't pause for CPU1